skytap/client.go

```diff
@@ -287,22 +359,147 @@ func (c *Client) checkResponse(r *http.Response) error {
 		errorResponse.RequestID = strToPtr(requestID)
 	}
 
-	if code := r.StatusCode; code == http.StatusLocked ||
-		code == http.StatusTooManyRequests ||
-		code == http.StatusConflict ||
-		(code >= http.StatusInternalServerError && code <= 599) {
-		if retryAfter := r.Header.Get(headerRetryAfter); retryAfter != "" {
-			val, err := strconv.Atoi(retryAfter)
-			if err == nil {
-				errorResponse.RetryAfter = intToPtr(val)
-			} else {
-				errorResponse.RetryAfter = intToPtr(c.retryAfter)
-			}
+	if retryAfter := r.Header.Get(headerRetryAfter); retryAfter != "" {
+		val, err := strconv.Atoi(retryAfter)
+		if err == nil {
+			errorResponse.RetryAfter = intToPtr(val)
 		} else {
 			errorResponse.RetryAfter = intToPtr(c.retryAfter)
 		}
-		errorResponse.RequiresRetry = true
+	} else {
+		errorResponse.RetryAfter = intToPtr(c.retryAfter)
 	}
 
 	return errorResponse
 }
+
+func (c *Client) checkResourceStateUntilSatisfied(ctx context.Context, req *http.Request, state *environmentVMRunState) error {
```

Comment:
This logic is very specific for the resource to be implemented at the HTTP client, consider to pass this function as an argument.

Reply:
Going to create a new ticket for this.